Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\lib\site-packages\jupyter_cache\executors\utils.py", line 51, in single_nb_execution
    executenb(
  File "C:\Users\randa\AppData\Roaming\Python\Python39\site-packages\nbclient\client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "C:\Users\randa\AppData\Roaming\Python\Python39\site-packages\nbclient\util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "C:\Users\randa\AppData\Roaming\Python\Python39\site-packages\nbclient\util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "C:\ProgramData\Anaconda3\lib\asyncio\base_events.py", line 642, in run_until_complete
    return future.result()
  File "C:\Users\randa\AppData\Roaming\Python\Python39\site-packages\nbclient\client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "C:\Users\randa\AppData\Roaming\Python\Python39\site-packages\nbclient\client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "C:\Users\randa\AppData\Roaming\Python\Python39\site-packages\nbclient\client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
S = firm.solve()
S.head()
------------------

[1;31m---------------------------------------------------------------------------[0m
[1;31mAttributeError[0m                            Traceback (most recent call last)
[1;32m~\AppData\Local\Temp\ipykernel_4388\2981563580.py[0m in [0;36m<cell line: 1>[1;34m()[0m
[1;32m----> 1[1;33m [0mS[0m [1;33m=[0m [0mfirm[0m[1;33m.[0m[0msolve[0m[1;33m([0m[1;33m)[0m[1;33m[0m[1;33m[0m[0m
[0m[0;32m      2[0m [0mS[0m[1;33m.[0m[0mhead[0m[1;33m([0m[1;33m)[0m[1;33m[0m[1;33m[0m[0m

[1;32mC:\ProgramData\Anaconda3\lib\site-packages\compecon\dpmodel.py[0m in [0;36msolve[1;34m(self, v, x, nr, **kwargs)[0m
[0;32m    456[0m             [0mself[0m[1;33m.[0m[0m__solve_by_function_iteration[0m[1;33m([0m[1;33m)[0m[1;33m[0m[1;33m[0m[0m
[0;32m    457[0m         [1;32melif[0m [0mself[0m[1;33m.[0m[0moptions[0m[1;33m.[0m[0malgorithm[0m [1;33m==[0m [1;34m'newton'[0m[1;33m:[0m[1;33m[0m[1;33m[0m[0m
[1;32m--> 458[1;33m             [0mself[0m[1;33m.[0m[0m__solve_by_Newton_method[0m[1;33m([0m[1;33m)[0m[1;33m[0m[1;33m[0m[0m
[0m[0;32m    459[0m         [1;32melse[0m[1;33m:[0m[1;33m[0m[1;33m[0m[0m
[0;32m    460[0m             [1;32mraise[0m [0mValueError[0m[1;33m([0m[1;34m'Unknown solution algorithm'[0m[1;33m)[0m[1;33m[0m[1;33m[0m[0m

[1;32mC:\ProgramData\Anaconda3\lib\site-packages\compecon\dpmodel.py[0m in [0;36m__solve_by_Newton_method[1;34m(self)[0m
[0;32m    827[0m             [0mcold[0m [1;33m=[0m [0mself[0m[1;33m.[0m[0mValue[0m[1;33m.[0m[0mc[0m[1;33m.[0m[0mcopy[0m[1;33m([0m[1;33m)[0m[1;33m.[0m[0mflatten[0m[1;33m([0m[1;33m)[0m[1;33m[0m[1;33m[0m[0m
[0;32m    828[0m             [1;31m# print('\ncold', cold)[0m[1;33m[0m[1;33m[0m[0m
[1;32m--> 829[1;33m             [0mself[0m[1;33m.[0m[0mValue_j[0m[1;33m[[0m[1;33m:[0m[1;33m][0m[1;33m,[0m [0mvc[0m [1;33m=[0m [0mself[0m[1;33m.[0m[0mvmax[0m[1;33m([0m[0ms[0m[1;33m,[0m [0mx[0m[1;33m,[0m [0mself[0m[1;33m.[0m[0mValue[0m[1;33m,[0m [1;32mTrue[0m[1;33m)[0m[1;33m[0m[1;33m[0m[0m
[0m[0;32m    830[0m             [0mself[0m[1;33m.[0m[0mmake_discrete_choice[0m[1;33m([0m[1;33m)[0m[1;33m[0m[1;33m[0m[0m
[0;32m    831[0m             [0mstep[0m [1;33m=[0m [1;33m-[0m [0mSOLVE[0m[1;33m([0m[0mPhik[0m [1;33m-[0m [0mvc[0m[1;33m,[0m [0mPhik[0m [1;33m@[0m [0mcold[0m [1;33m-[0m [0mself[0m[1;33m.[0m[0mValue[0m[1;33m.[0m[0my[0m[1;33m.[0m[0mflatten[0m[1;33m([0m[1;33m)[0m[1;33m)[0m[1;33m[0m[1;33m[0m[0m

[1;32mC:\ProgramData\Anaconda3\lib\site-packages\compecon\dpmodel.py[0m in [0;36mvmax[1;34m(self, s, x, Value, dVc)[0m
[0;32m    894[0m                                 [0msnext[0m [1;33m=[0m [0mself[0m[1;33m.[0m[0mtransition[0m[1;33m([0m[0ms[0m[1;33m[[0m[1;33m:[0m[1;33m,[0m [0mis_[0m[1;33m][0m[1;33m,[0m [0mx[0m[1;33m[[0m[0mi[0m[1;33m,[0m [0mj[0m[1;33m,[0m [1;33m:[0m[1;33m,[0m [0mis_[0m[1;33m][0m[1;33m,[0m [0mi[0m [1;33m,[0m [0mj[0m[1;33m,[0m [0min_[0m[1;33m,[0m [0mee[0m[1;33m[[0m[1;33m:[0m[1;33m,[0m [0mis_[0m[1;33m][0m[1;33m)[0m  [1;31m#fixme need to know number of output arguments!!![0m[1;33m[0m[1;33m[0m[0m
[0;32m    895[0m                                 [0mprob[0m [1;33m=[0m [0mw[0m[1;33m[[0m[0mk[0m[1;33m][0m [1;33m*[0m [0mq[0m[1;33m[[0m[0mj[0m[1;33m,[0m [0mi[0m[1;33m,[0m [0min_[0m[1;33m,[0m[1;33m][0m[1;33m[0m[1;33m[0m[0m
[1;32m--> 896[1;33m                                 [0mvc[0m[1;33m[[0m[0mis_[0m[1;33m,[0m [0mi[0m[1;33m,[0m [1;33m:[0m[1;33m,[0m [0min_[0m[1;33m][0m [1;33m+=[0m [0mprob[0m [1;33m*[0m [0mValue[0m[1;33m.[0m[0mPhi[0m[1;33m([0m[0msnext[0m[1;33m)[0m[1;33m.[0m[0mtoarray[0m[1;33m([0m[1;33m)[0m[1;33m.[0m[0mreshape[0m[1;33m([0m[1;33m([0m[0mis_[0m[1;33m.[0m[0msum[0m[1;33m([0m[1;33m)[0m[1;33m,[0m [0mms[0m[1;33m)[0m[1;33m,[0m [0morder[0m[1;33m=[0m[1;34m'F'[0m[1;33m)[0m   [1;31m#fixme I can't find the proper way to index this[0m[1;33m[0m[1;33m[0m[0m
[0m[0;32m    897[0m [1;33m[0m[0m
[0;32m    898[0m             [0mvc[0m [1;33m=[0m [0mvc[0m[1;33m.[0m[0mreshape[0m[1;33m([0m[1;33m([0m[0mns[0m[1;33m*[0m[0mni[0m[1;33m,[0m[0mms[0m[1;33m*[0m[0mni[0m[1;33m)[0m[1;33m,[0m[0morder[0m[1;33m=[0m[1;34m'F'[0m[1;33m)[0m[1;33m[0m[1;33m[0m[0m

[1;31mAttributeError[0m: 'numpy.ndarray' object has no attribute 'toarray'
AttributeError: 'numpy.ndarray' object has no attribute 'toarray'

